src/JuiceDelegatesRegistry.sol

```diff
@@ -1,6 +1,44 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.17;
 
+import '@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBPayDelegate.sol';
+import '@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBRedemptionDelegate.sol';
+import '@openzeppelin/contracts/utils/introspection/ERC165Checker.sol';
+
 contract JuiceDelegatesRegistry {
+    error juiceDelegatesRegistry_incompatibleDelegate();
+    error juiceDelegatesRegistry_authError();
+
+    mapping(address => bool) public trustedDeployers;
+
+    // delegate => deployer
+    mapping(address => address) public trustedJuiceDelegates;
+
+    mapping(address => bool) public admins;
+
+    modifier onlyDeployer() {
+        if(!trustedDeployers[msg.sender]) revert juiceDelegatesRegistry_authError();
+        _;
+    }
+
```

Comment:
pretty neat :)
things to be added
- we need a constructor setting the initial admin
- remove admin, deployer & delegate methods